File: bun-elasticemail-examples/javascript/examples/contacts.js
import { Configuration, ContactsApi, ListsApi } from "@elasticemail/elasticemail-client-ts-axios";

const config = new Configuration({ apiKey: process.env.ELASTICEMAIL_API_KEY });
const contactsApi = new ContactsApi(config);
const listsApi = new ListsApi(config);

const listName = process.env.ELASTICEMAIL_LIST_NAME || "Newsletter";
const email = process.env.EMAIL_TO || "you@yourdomain.com";

const fail = (step, err) => {
  console.error(`Error (${step}):`, err.response?.status, err.response?.data ?? err.message);
  process.exit(1);
};

// 1. Create a list (Elastic Email lists are addressed by name, not by id)
try {
  await listsApi.listsPost({ ListName: listName, AllowUnsubscribe: true });
  console.log(`List "${listName}" created.`);
} catch (err) {
  if (err.response?.status === 400 && /exist/i.test(JSON.stringify(err.response?.data))) {
    console.log(`List "${listName}" already exists.`);
  } else {
    fail("create list", err);
  }
}

// 2. Add a contact and put it on the list in one call
try {
  const { data } = await contactsApi.contactsPost(
    [
      {
        Email: email,
        FirstName: "Ann",
        LastName: "Example",
        Status: "Active",
        CustomFields: { plan: "Pro" }, // only existing custom fields are saved
      },
    ],
    [listName],
  );
  console.log("Contact added:", data[0]?.Email, "status:", data[0]?.Status);
} catch (err) {
  fail("add contact", err);
}

// 3. Read it back
try {
  const { data } = await contactsApi.contactsByEmailGet(email);
  console.log("Contact:", { Email: data.Email, FirstName: data.FirstName, Status: data.Status, Source: data.Source });
} catch (err) {
  fail("get contact", err);
}

// 4. Update
try {
  const { data } = await contactsApi.contactsByEmailPut(email, { FirstName: "Anna" });
  console.log("Contact updated. FirstName:", data.FirstName);
} catch (err) {
  fail("update contact", err);
}

// 5. List contacts on the list
try {
  const { data } = await listsApi.listsByListnameContactsGet(listName, 10, 0);
  console.log(`Contacts in "${listName}" (first ${data.length}):`);
  for (const c of data) console.log(" -", c.Email, c.Status);
} catch (err) {
  fail("list contacts", err);
}

// 6. Delete the contact (uncomment to clean up)
// await contactsApi.contactsByEmailDelete(email);
// console.log("Contact deleted.");

console.log("\nDone.");

File: rust-elasticemail-examples/examples/contacts.rs
use rust_elasticemail_examples::{already_exists, config, env_or, print_api_error, to};
use std::collections::HashMap;
use ElasticEmail::apis::{contacts_api, lists_api};
use ElasticEmail::models::{ContactPayload, ContactStatus, ContactUpdatePayload, ListPayload};

#[tokio::main]
async fn main() {
    let config = config();
    let list_name = env_or("ELASTICEMAIL_LIST_NAME", "Newsletter");
    let email = to();

    // 1. Create a list (Elastic Email lists are addressed by name, not by id)
    let list = ListPayload { allow_unsubscribe: Some(true), ..ListPayload::new(list_name.clone()) };
    match lists_api::lists_post(&config, list).await {
        Ok(_) => println!("List \"{}\" created.", list_name),
        Err(e) if already_exists(&e) => println!("List \"{}\" already exists.", list_name),
        Err(e) => {
            print_api_error("create list", &e);
            std::process::exit(1);
        }
    }

    // 2. Add a contact and put it on the list in one call
    let payload = ContactPayload {
        first_name: Some("Ann".to_string()),
        last_name: Some("Example".to_string()),
        status: Some(ContactStatus::Active),
        // only existing custom fields are saved
        custom_fields: Some(HashMap::from([("plan".to_string(), "Pro".to_string())])),
        ..ContactPayload::new(email.clone())
    };
    match contacts_api::contacts_post(&config, vec![payload], Some(vec![list_name.clone()])).await {
        Ok(contacts) => {
            let c = contacts.first();
            println!(
                "Contact added: {} status: {:?}",
                c.and_then(|c| c.email.clone()).unwrap_or_default(),
                c.and_then(|c| c.status)
            );
        }
        Err(e) => {
            print_api_error("add contact", &e);
            std::process::exit(1);
        }
    }

    // 3. Read it back
    match contacts_api::contacts_by_email_get(&config, &email).await {
        Ok(c) => println!(
            "Contact: email={} first_name={} status={:?} source={:?}",
            c.email.unwrap_or_default(),
            c.first_name.unwrap_or_default(),
            c.status,
            c.source
        ),
        Err(e) => {
            print_api_error("get contact", &e);
            std::process::exit(1);
        }
    }

    // 4. Update
    let update = ContactUpdatePayload { first_name: Some("Anna".to_string()), ..ContactUpdatePayload::new() };
    match contacts_api::contacts_by_email_put(&config, &email, update).await {
        Ok(c) => println!("Contact updated. FirstName: {}", c.first_name.unwrap_or_default()),
        Err(e) => {
            print_api_error("update contact", &e);
            std::process::exit(1);
        }
    }

    // 5. List contacts on the list
    match lists_api::lists_by_listname_contacts_get(&config, &list_name, Some(10), Some(0)).await {
        Ok(contacts) => {
            println!("Contacts in \"{}\" (first {}):", list_name, contacts.len());
            for c in contacts {
                println!(" - {} {:?}", c.email.unwrap_or_default(), c.status);
            }
        }
        Err(e) => {
            print_api_error("list contacts", &e);
            std::process::exit(1);
        }
    }

    // 6. Delete the contact (uncomment to clean up)
    // contacts_api::contacts_by_email_delete(&config, &email).await.unwrap();
    // println!("Contact deleted.");

    println!("\nDone.");
}
